projects/wave-core-new/src/lib/operators/operator-type-parameter-options.model.ts

```diff
@@ -0,0 +1,484 @@
+
+    constructor(config: DictParameterArrayConfig<T>) {
+        super();
+        this.options = config.options;
+    }
+
+    get parameterType(): ParameterOptionType {
+        return ParameterOptionType.DICT_ARRAY;
+    }
+
+    static fromDict<T extends OptionsDict>(dict: DictParameterArrayConfig<T>): DictParameterArray<T> {
+        return new DictParameterArray(dict);
+    }
+
+    toDict(): DictParameterArrayConfig<T> {
+        return {
+            options: this.options,
+            kind: ParameterOptionType.DICT_ARRAY,
+        };
+    }
+
+    get firstTick(): number {
+        return 0;
+    }
+
+    getDisplayValueForTick(tick: number): string | undefined {
+        return this.getValueForTick(tick).displayValue.toString();
+    }
+
+    getTickForValue(value: T): number | undefined {
+        const findIndex = this.options.findIndex(x => x.displayValue === value.displayValue); // TODO: review if displayValue is distinct
+        if (findIndex < 0) {
+            return undefined;
+        }
+        return findIndex;
+    }
+
+    getValueForTick(tick: number): T | undefined {
+        if (!this.isValidTick(tick)) {
+            return undefined;
+        }
+        return this.options[tick];
+    }
+
+    get lastTick(): number {
+        return this.options.length - 1;
+    }
+
+    get tickType(): TickType {
+        return TickType.DISCRETE;
+    }
+}
+
+
+/**
+ * Dictionary for serializing the operator type.
+ */
+export interface OperatorTypeParameterOptionsConfig {
+    operatorType: string;
+}
+
+/**
+ * The operator basic type.
+ */
+export abstract class OperatorTypeParameterOptions {
+    operatorType: string;
+
+    constructor(config: OperatorTypeParameterOptionsConfig) {
+        this.operatorType = config.operatorType;
+    }
+
+    /**
+     * Serialize the operator type.
+     */
+    abstract toDict(): OperatorTypeParameterOptionsConfig;
+
+    /**
+     * Get the parameter option type for a parameter name.
+     */
+    public getParameterOption(parameterName: ParameterName): ParameterContainerType {
+        const res: [ParameterName, ParameterContainerType] = this.getParameterOptions().find(
+            ([pN, _]) => pN === parameterName
+        );
+        if (!res) {
+            return AbstractParameterContainer.getEmptyOption();
+        }
+        const [_, pOption] = res;
+        return pOption;
+    }
+
+    /**
+     * Get the parameter options as array of tuples [ParameterName, ParameterContainerType].
+     */
+    public abstract getParameterOptions(): Array<[ParameterName, ParameterContainerType]>;
+
+    /**
+     * Get the parameter options as array of tuples [ParameterName, ParameterOptionType].
+     */
+    public abstract getParametersTypes(): Array<[ParameterName, ParameterOptionType]>;
+
+    /**
+     * Get a list of all parameter names.
+     */
+    public getParameterNames(): Array<ParameterName> {
+        return this.getParametersTypes().map(([n, _]) => n);
+    }
+}
```

Comment:
I guess this all doesn't work currently :disappointed: .

Reply:
no. should I remove it for now?

Reply:
I guess so.

Reply:
a05473d8cac2769f869b74539507f8e6eb972f27